### DISCUSSION TITLE
BootStrap 5 breakpoints vs Vanjaro previewer breakpoints vs. Vanjaro Image Optimizer breakpoints

### DISCUSSION THREAD

**Discussion by X3Technology** [vanjarosoftware/Vanjaro.Platform#892] [General]
BootStrap 5 has 6 breakpoints: xs>0px, sm>576px, md>768px, lg>992px, xl>1200px, xxl>1400px.   
The Vanjaro preview function does mobile=360px, tablet=768px, and desktop .  
The Vanjaro Image optimizer is saving 5 different images sizes (360px, 720px, 1280px, 1920px and original) in both webp and jpeg.  

Why the different number of breakpoint tiers and pixels?

Would it make sense to have a preview function for all 6 BootStrap breakpoints?
And to have optimized images for each BS5 breakpoint as well?

**Comment by meetmandeep** [comment]
### Responsive Mode
There are 3 different **Device Views** in Vanjaro. Here's how they correspond to Bootstrap, Media Queries, and Preview. 

#### Desktop
The screen width preview is 100%
Bootstrap breakpoint "lg" in Grid (.col-lg)

#### Tablet
The screen width preview is 768px
Bootstrap breakpoint "sm" is used in Grid (.col-sm)
All other styling using max-width: 991px

#### Mobile
The screen width preview is 360px
No Bootstrap breakpoint is used in Grid (.col)
All other styling using max-width: 576px

### Vanjaro Image Optimizer 

>The Vanjaro Image optimizer is saving 5 different images sizes (360px, 720px, 1280px, 1920px and original) in both webp and jpeg.

Images are specifically saved in these different formats to handle devices across all viewports including Mobile, Tables, Desktops, and Ultra Desktops. It's important to understand that image dimension has very little to do with the viewport width; rather, the actual size the image is shown on the page. For example, we could have an image that is 120px wide displayed on a mobile device. In this use case, 360px is the correct size as long as the device's DPR (Device Pixel Ratio) is 3 or less. Vanjaro will automatically serve a 720px image if the DPR is greater than 3. 


### Other Questions

>Would it make sense to have a preview function for all 6 BootStrap breakpoints?

Not really. Unless you have a specific requirement, you can easily build a site for mobile all the way up to Ultra Desktops. 

>And to have optimized images for each BS5 breakpoint as well?

No. Refer to Vanjaro Image Optimizer explanation above.

**Reply by X3Technology** [reply]
Excellent explanation.  Thanks @meetmandeep !
